test/Sentry.Tests/SentrySdkTests.cs

```diff
@@ -831,6 +843,82 @@ public void InitHub_DebugEnabled_DebugLogsLogged()
             Arg.Any<object[]>());
     }
 
+    #if __IOS__
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void ProcessOnBeforeSend_SegfaultsSuppressed(bool suppressSegfaults)
+    {
+        var options = new SentryOptions
+        {
+            Dsn = ValidDsn,
+            DiagnosticLogger = _logger,
+            IsGlobalModeEnabled = true,
+            Debug = true,
+            AutoSessionTracking = false,
+            BackgroundWorker = Substitute.For<IBackgroundWorker>(),
+            InitNativeSdks = false
+        };
+        options.Native.SuppressSegfaults = suppressSegfaults;
+
+        var evt = new Sentry.CocoaSdk.SentryEvent();
+        var ex = new Sentry.CocoaSdk.SentryException("Signal 6, Code 0", "SIGABRT");
+        var st = new Sentry.CocoaSdk.SentryStacktrace(
+            [new Sentry.CocoaSdk.SentryFrame
+            {
+                Function = "xamarin_unhandled_exception_handler"
+            }],
+            new Foundation.NSDictionary<Foundation.NSString, Foundation.NSString>()
+        );
+        ex.Stacktrace = st;
+        evt.Exceptions = [ex];
+        var result = SentrySdk.ProcessOnBeforeSend(options, evt);
+
+        if (suppressSegfaults)
+        {
+            result.Should().BeNull();
```

Comment:
We could also ensure any custom beforesend handler isn't called in this scenario.

Reply:
Done